internal/bminventory/inventory.go

```diff
@@ -1653,6 +1660,23 @@ func (b *bareMetalInventory) UpdateClusterNonInteractive(ctx context.Context, pa
 	return b.v2UpdateClusterInternal(ctx, params, NonInteractive)
 }
 
+func updateClusterNonePlatformParams(params installer.V2UpdateClusterParams, cluster *common.Cluster) {
+	if params.ClusterUpdateParams.UserManagedNetworking == nil || cluster.Platform == nil || cluster.UserManagedNetworking == nil {
```

Comment:
This should be && no?

Reply:
No, If one of those properties is not set it will not be able to update the platform/UMN

Reply:
Hmmm, unsure I understand the logic here.
Don't we have a default for `cluster.Platform`? under what circumstances it will be nil?
Same question for `cluster.UserManagedNetworking`.

If `params.ClusterUpdateParams.UserManagedNetworking` is nil but `params.ClusterUpdateParams.Platform.Type == models.PlatformTypeNone` you would want to update `params.ClusterUpdateParams.UserManagedNetworking` to `swag.Bool(true)` no?

Reply:
@eranco74 You are right, I removed all the unnecessary nil "validations"